src/Microsoft.Dnx.Runtime/FrameworkReferenceResolver.cs

```diff
@@ -277,7 +342,7 @@ private static FrameworkInformation GetFrameworkInformation(DirectoryInfo direct
 
                             var entry = new AssemblyEntry();
                             entry.Version = version != null ? Version.Parse(version) : null;
-                            frameworkInfo.Assemblies.Add(assemblyName, entry);
+                            frameworkInfo.Assemblies[assemblyName] = entry;
```

Comment:
added one. Microsoft.Build.Engine is present in both the 2.0 and 3.5 folders (thus the 3.5 one should override it in our logic, I'm confirming that this is the behavior for MSBuild)